npm/podController.go

```diff
@@ -465,40 +498,36 @@ func (c *podController) syncAddAndUpdatePod(newPodObj *corev1.Pod) error {
 		// new PodIP needs to be added to all newLabels
 		addToIPSets = util.GetIPSetListFromLabels(newPodObj.Labels)
 
-		log.Logf("Deleting pod %s %s from ipset %s and adding pod %s to ipset %s",
+		klog.Infof("Deleting pod %s %s from ipset %s and adding pod %s to ipset %s",
 			cachedNpmPodObj.PodUID, cachedNpmPodObj.PodIP, cachedNpmPodObj.Namespace, newPodObj.Status.PodIP, newPodObjNs,
 		)
 
 		// Delete the pod from its namespace's ipset.
 		if err = ipsMgr.DeleteFromSet(cachedNpmPodObj.Namespace, cachedNpmPodObj.PodIP, podKey); err != nil {
```

Comment:
This looks a bug. We need to append "ns-".
if err = ipsMgr.DeleteFromSet(cachedNpmPodObj.Namespace, cachedNpmPodObj.PodIP, podKey); err != nil 
-> if err = ipsMgr.DeleteFromSet(util.GetNSNameWithPrefix(cachedNpmPodObj.Namespace) , cachedNpmPodObj.PodIP, podKey); err != nil 

We need to improve UT to check real ipset as well to catch this bug.